### PR TITLE
Refactor `SetupVethPair` to prepare for reuse with OCI containers

### PR DESCRIPTION
**Pull request by bduffany**
* Remove `vmIdx` parameter of `SetupVethPair` - it is only used to keep track of the next host IP that we should try to reserve. This can instead be an internal implementation detail of `HostNetAllocator`, instead of relying on the caller to "hint" at what the next "likely free" index will be.
* Remove TAP device configuration from `CreateVethPair` and into a separate function (more context/discussion to follow in Slack)

**Related issues**: N/A

### CODE REVIEW

**Review by tylerwilliams — APPROVED** [review]
LGTM but getting a review from @vadimberezniker and @maggie-lou too probably wouldn't hurt :)

File: server/util/networking/networking.go [hunk 1]
```diff
@@ -236,9 +240,12 @@ func (n *HostNet) Unlock() {
 }
 
 // Get assigns a host network IP for the given VM index.
-func (a *HostNetAllocator) Get(vmIdx int) (*HostNet, error) {
+func (a *HostNetAllocator) Get() (*HostNet, error) {
+	a.mu.Lock()
+	defer a.mu.Unlock()
 	for attempt := 0; attempt < numAssignableNetworks; attempt++ {
-		netIdx := (vmIdx + attempt) % numAssignableNetworks
+		netIdx := a.idx % numAssignableNetworks
```
Comment (maggie-lou) [inline server/util/networking/networking.go:247]: nit: don't need to % numAssignableNetworks if you're doing it in the line below

File: server/util/networking/networking.go [hunk 2]
```diff
@@ -236,9 +240,12 @@ func (n *HostNet) Unlock() {
...
+		a.idx = (a.idx + 1) % numAssignableNetworks
 		if a.tryLock(netIdx) {
```
Comment (maggie-lou) [inline server/util/networking/networking.go:249]: nit: Maybe rename this given we're not locking anymore?

**Review by maggie-lou — APPROVED** [review]

File: server/util/networking/networking.go (hunk 1 above)
Comment (bduffany) [reply server/util/networking/networking.go]: good catch, removed

File: server/util/networking/networking.go (hunk 2 above)
Comment (bduffany) [reply server/util/networking/networking.go]: I just inlined the logic to avoid confusion

**Review by vadimberezniker — APPROVED** [review]

**Pull request merged**